x-pack/plugins/security/public/management/roles/edit_role/edit_role_page.tsx

```diff
@@ -221,14 +220,21 @@ function useRole(
   return [role, setRole] as [Role | null, typeof setRole];
 }
 
-function useSpaces(http: HttpStart, fatalErrors: FatalErrorsSetup, spacesEnabled: boolean) {
-  const [spaces, setSpaces] = useState<Space[] | null>(null);
+function useSpaces(http: HttpStart, fatalErrors: FatalErrorsSetup) {
+  const [spaces, setSpaces] = useState<{ enabled: boolean; list: Space[] } | null>(null);
   useEffect(() => {
-    (spacesEnabled ? http.get('/api/spaces/space') : Promise.resolve([])).then(
-      fetchedSpaces => setSpaces(fetchedSpaces),
-      err => fatalErrors.add(err)
+    http.get('/api/spaces/space').then(
+      fetchedSpaces => setSpaces({ enabled: true, list: fetchedSpaces }),
+      (err: IHttpFetchError) => {
+        // Spaces plugin can be disabled and hence this endpoint can be unavailable.
+        if (err.response?.status === 404) {
```

Comment:
__note:__ I moved this change to a separate commit as I want to hear your thoughts. It's "the least evil" solution I could come up with. Alternatively we can introduce an endpoint that will return `isSpacesEnabled` only based on the "Spaces Service" that Spaces registers with Security during setup.

Reply:
I think this is tolerable. I think it'd make sense (in a followup) for us to export a `useSpaces` hook from the Spaces plugin itself, so that we could at least have the Spaces plugin own this determination.

The disabled check aside, I could see other plugins benefiting from this hook too.

Reply:
Yeah, I agree